app/src/main/java/org/elasticsearch/bootstrap/BootstrapProxy.java

```diff
@@ -104,8 +102,6 @@ public class BootstrapProxy {
             }
         }, "crate[keepAlive/" + Version.CURRENT + "]");
         keepAliveThread.setDaemon(false);
-        // keep this thread alive (non daemon thread) until we shutdown
-        Runtime.getRuntime().addShutdownHook(new Thread(keepAliveLatch::countDown));
     }
```

Comment:
I guess we can remove it since we have _always_ add a hook in setup now. 

But looking into `BootstrapProxy.init`, there is some code which can fail before `INSTANCE.setup` call, ie hook won't be added. Is it okay?

`LogConfigurator.configure(environment)` or `PidFile.create(environment.pidFile(), true)` can throw before we do setup/add a hook.